### PR TITLE
Fix console colors for non-windows platforms

### PR DESCRIPTION
**Pull request by kichalla**
@rynowak 

image

### CODE REVIEW

**Comment by rynowak** [comment]
@kichalla - where's the gray?

File: src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs [hunk 1]
```diff
@@ -16,6 +17,9 @@ public class ConsoleLogger : ILogger
         private static readonly string _loglevelPadding = ": ";
         private static readonly string _messagePadding;
 
+        // ConsoleColor does not have a value to specify the 'Default' color
+        private readonly ConsoleColor? DEFAULT_CONSOLE_COLOR = null;
```
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs]: `DEFAULT_CONSOLE_COLOR` 

???

File: src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        {
+            System.Console.Write(_outputBuilder.ToString());
+            _outputBuilder.Clear();
+        }
+
+        private string SetForegroundColor(string message, ConsoleColor? foreground)
+        {
+            if (foreground.HasValue)
+            {
+                return GetForegroundColorEscapeCode(foreground.Value) + message + "\x1B[39m";
```
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs]: is `"\x1B[39m"` the code for 'reset this'?

File: src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            _outputBuilder.AppendLine();
+        }
+
+        public void Flush()
+        {
+            System.Console.Write(_outputBuilder.ToString());
+            _outputBuilder.Clear();
+        }
+
+        private string SetForegroundColor(string message, ConsoleColor? foreground)
```
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs]: You want to write these things to the builder instead of appending them as strings.

File: src/Microsoft.Extensions.Logging.Console/Internal/LogConsole.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    public class LogConsole : IConsole
```
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/LogConsole.cs]: `WindowsLogConsole`?

File: test/Microsoft.Extensions.Logging.Test/Console/TestConsole.cs [hunk 5]
```diff
@@ -8,39 +8,55 @@ namespace Microsoft.Extensions.Logging.Test.Console
 {
     public class TestConsole : IConsole
     {
-        public const ConsoleColor DefaultBackgroundColor = ConsoleColor.Black;
-        public const ConsoleColor DefaultForegroundColor = ConsoleColor.Gray;
+        public static readonly ConsoleColor? DefaultBackgroundColor = null;
+        public static readonly ConsoleColor? DefaultForegroundColor = null;
```
Comment (rynowak) [inline test/Microsoft.Extensions.Logging.Test/Console/TestConsole.cs]: `= null` isn't needed

File: src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs]: coding standard bro :sunglasses:

File: src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs (hunk 2 above)
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/AnsiLogConsole.cs]: Might want to just leave a comment what the codes mean when it's not obvious.

**Comment by rynowak** [comment]
:watch: - this is looking pretty good.

**Comment by kichalla** [comment]
Updated @rynowak

File: src/Microsoft.Extensions.Logging.Console/Internal/IAnsiSystemConsole.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    public interface IAnsiSystemConsole
```
Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/IAnsiSystemConsole.cs]: Where did this come from? Why is this needed?

Comment (kichalla) [inline src/Microsoft.Extensions.Logging.Console/Internal/IAnsiSystemConsole.cs]: yeah I know...its more 'console' interfaces...I wanted to add tests for AnsiLogConsole as it has considerable amount of code...

Comment (rynowak) [inline src/Microsoft.Extensions.Logging.Console/Internal/IAnsiSystemConsole.cs]: I don't see the tests in the PR...

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
7fe4239da5fdbf3d4a555f85bca0c91621d126e9
